src/policy/feerate.cpp

```diff
@@ -18,6 +17,9 @@ CFeeRate::CFeeRate(const CAmount& nFeePaid, size_t nBytes_)
         nSatoshisPerK = nFeePaid * 1000 / nSize;
     else
         nSatoshisPerK = 0;
+
+    if (nSatoshisPerK < MIN_RELAY_TX_FEE)
```

Comment:
I haven't looked at all the places where this constructor is used, but I'm worried this change might cause some unwanted side effects.

Reply:
I think we should look at all the places where this constructor is used, and adjust the code for the constructor accordingly. It may be that we need to change line 17 (nSatoshisPerK = nFeePaid * 1000 / nSize;)

Reply:
I took a look at all places. There is 3 way to instanciate a CFeeRate:
- no params: it will get a nSatoshiPerK at 0.2
- 1 params: the nSatoshi will be define based on the given Params. I didn't change anything for that one because it's used in multiple part of the code to allow the user to overide the min fee (but there is always a check to validate the min)
- 2 params: here we are :) i put that check for safety reason but i didn't find case that can be enteting here (if i remember correctly. I'm off without computer for a few days :p)

Reply:
0 and 1 params are probably fine, but the constructor for 2 params is tricky.

Look at the code in line 17:
`        nSatoshisPerK = nFeePaid * 1000 / nSize;`

Let's say nSize = 1, and nFeePaid = MIN_RELAY_TX_FEE. Then nSatoshisPerK = MIN_RELAY_TX_FEE*1000. Is that what we want here?

Reply:
What about we just ignore the nSize and do something like
`
    if (nSize > 0)
        nSatoshisPerK = nFeePaid;
    else
        nSatoshisPerK = 0;
`
Of course we keep the check of minimum 0.2 for security

Reply:
Maybe. It depends how CFeeRate is being used exactly. Are the uses where nSatoshisPerK should be something different from 0.2?

Reply:
In Verium case, that constructor (with 2 params)  is mainly use for the UI & the args pass to the binary to override default fee (in case a user want to pay more :D)